torch/_decomp/__init__.py

```diff
@@ -375,6 +375,7 @@ def core_aten_decompositions() -> Dict[torch._ops.OperatorBase, Callable]:
             aten.rad2deg_,
             aten.renorm,
             aten.renorm_,
+            aten.replication_pad2d,
```

Comment:
Will need to benchmark this, I suspect inductor won't generate anything reasonable here,

Reply:
What exactly should I benchmark? Should I do something like the following and then just call the compiled function with a range of different inputs?

```python
def f0(input, padding):
    return torch.ops.aten.replication_pad2d.default(input, padding)

f0_opt = torch.compile(f0)
```

Reply:
Or would it be better to chain multiple calls like this?

```python
def f0(input, padding):
    res = torch.ops.aten.replication_pad2d.default(input, padding)
    for _ in range(N - 1):
        res = torch.ops.aten.replication_pad2d.default(res, padding)
    return res 
```

Reply:
I have a performance measurement script here: https://github.com/kurtamohler/pytorch-perf-test-scripts/blob/7d84beaf4813a00ae57b3c22d62238951f6658eb/replication_pad2d/decomp_perf.py

It just measures one call to `torch.ops.aten.replication_pad2d.default`, eager vs compiled, with a bunch of different inputs.

Without the decomp from this PR, I get the following measurements:

<details><summary>Click to expand</summary>

```
input_size, padding, device, dtype, eager_time, compile_time, speedup
(10, 10, 10), (2, 2, 2, 2), cpu, torch.float32, 0.0000167, 0.0000506, 0.329
(10, 10, 10), (2, 2, 2, 2), cpu, torch.float64, 0.0000176, 0.0000512, 0.345
(10, 10, 10), (2, 2, 2, 2), cpu, torch.int32, 0.0000165, 0.0000506, 0.326
(10, 10, 10), (2, 2, 2, 2), cuda, torch.float32, 0.0000229, 0.0000629, 0.365
(10, 10, 10), (2, 2, 2, 2), cuda, torch.float64, 0.0000229, 0.0000629, 0.364
(10, 10, 10), (2, 2, 2, 2), cuda, torch.int32, 0.0000230, 0.0000626, 0.368
(1000, 10, 10), (2, 2, 2, 2), cpu, torch.float32, 0.0001392, 0.0001740, 0.800
(1000, 10, 10), (2, 2, 2, 2), cpu, torch.float64, 0.0001475, 0.0001843, 0.800
(1000, 10, 10), (2, 2, 2, 2), cpu, torch.int32, 0.0001399, 0.0001741, 0.803
(1000, 10, 10), (2, 2, 2, 2), cuda, torch.float32, 0.0000233, 0.0000623, 0.375
(1000, 10, 10), (2, 2, 2, 2), cuda, torch.float64, 0.0000232, 0.0000609, 0.380
(1000, 10, 10), (2, 2, 2, 2), cuda, torch.int32, 0.0000232, 0.0000616, 0.377
(10, 100, 100), (10, 10, 10, 10), cpu, torch.float32, 0.0000414, 0.0000755, 0.549
(10, 100, 100), (10, 10, 10, 10), cpu, torch.float64, 0.0000473, 0.0000818, 0.579
(10, 100, 100), (10, 10, 10, 10), cpu, torch.int32, 0.0000418, 0.0000758, 0.551
(10, 100, 100), (10, 10, 10, 10), cuda, torch.float32, 0.0000228, 0.0000616, 0.369
(10, 100, 100), (10, 10, 10, 10), cuda, torch.float64, 0.0000232, 0.0000614, 0.377
(10, 100, 100), (10, 10, 10, 10), cuda, torch.int32, 0.0000229, 0.0000614, 0.372
(10, 100, 100), (100, 100, 100, 100), cpu, torch.float32, 0.0004629, 0.0005015, 0.923
(10, 100, 100), (100, 100, 100, 100), cpu, torch.float64, 0.0004967, 0.0005386, 0.922
(10, 100, 100), (100, 100, 100, 100), cpu, torch.int32, 0.0004075, 0.0004401, 0.926
(10, 100, 100), (100, 100, 100, 100), cuda, torch.float32, 0.0000423, 0.0000696, 0.608
(10, 100, 100), (100, 100, 100, 100), cuda, torch.float64, 0.0000443, 0.0000717, 0.619
(10, 100, 100), (100, 100, 100, 100), cuda, torch.int32, 0.0000423, 0.0000696, 0.608
(100, 100, 100), (10, 10, 10, 10), cpu, torch.float32, 0.0004556, 0.0005044, 0.903
(100, 100, 100), (10, 10, 10, 10), cpu, torch.float64, 0.0010030, 0.0010874, 0.922
(100, 100, 100), (10, 10, 10, 10), cpu, torch.int32, 0.0004420, 0.0004899, 0.902
(100, 100, 100), (10, 10, 10, 10), cuda, torch.float32, 0.0000617, 0.0000914, 0.676
(100, 100, 100), (10, 10, 10, 10), cuda, torch.float64, 0.0000696, 0.0000969, 0.719
(100, 100, 100), (10, 10, 10, 10), cuda, torch.int32, 0.0000614, 0.0000904, 0.680
(10, 1000, 1000), (10, 10, 10, 10), cpu, torch.float32, 0.0105431, 0.0109498, 0.963
(10, 1000, 1000), (10, 10, 10, 10), cpu, torch.float64, 0.0211419, 0.0220613, 0.958
(10, 1000, 1000), (10, 10, 10, 10), cpu, torch.int32, 0.0107333, 0.0149306, 0.719
(10, 1000, 1000), (10, 10, 10, 10), cuda, torch.float32, 0.0003771, 0.0003770, 1.000
(10, 1000, 1000), (10, 10, 10, 10), cuda, torch.float64, 0.0005622, 0.0005632, 0.998
(10, 1000, 1000), (10, 10, 10, 10), cuda, torch.int32, 0.0002965, 0.0002962, 1.001
(10, 1000, 1000), (100, 100, 100, 100), cpu, torch.float32, 0.0162436, 0.0165354, 0.982
(10, 1000, 1000), (100, 100, 100, 100), cpu, torch.float64, 0.0304805, 0.0312891, 0.974
(10, 1000, 1000), (100, 100, 100, 100), cpu, torch.int32, 0.0165726, 0.0168720, 0.982
(10, 1000, 1000), (100, 100, 100, 100), cuda, torch.float32, 0.0005086, 0.0004669, 1.089
(10, 1000, 1000), (100, 100, 100, 100), cuda, torch.float64, 0.0006622, 0.0006711, 0.987
(10, 1000, 1000), (100, 100, 100, 100), cuda, torch.int32, 0.0003875, 0.0003873, 1.001
```

</details>

With the decomp from this PR, I get the following. I've marked some of the rows with "*" to indicate that their compiled performance is significantly worse than the previous results.

<details><summary>Click to expand</summary>

```
input_size, padding, device, dtype, eager_time, compile_time, speedup
(10, 10, 10), (2, 2, 2, 2), cpu, torch.float32, 0.0000167, 0.0000471, 0.354
(10, 10, 10), (2, 2, 2, 2), cpu, torch.float64, 0.0000168, 0.0000471, 0.358
(10, 10, 10), (2, 2, 2, 2), cpu, torch.int32, 0.0000166, 0.0000471, 0.353
(10, 10, 10), (2, 2, 2, 2), cuda, torch.float32, 0.0000330, 0.0000980, 0.337
(10, 10, 10), (2, 2, 2, 2), cuda, torch.float64, 0.0000225, 0.0000655, 0.344
(10, 10, 10), (2, 2, 2, 2), cuda, torch.int32, 0.0000225, 0.0000651, 0.346
* (1000, 10, 10), (2, 2, 2, 2), cpu, torch.float32, 0.0001353, 0.0002456, 0.551
* (1000, 10, 10), (2, 2, 2, 2), cpu, torch.float64, 0.0001454, 0.0002335, 0.623
* (1000, 10, 10), (2, 2, 2, 2), cpu, torch.int32, 0.0001391, 0.0002320, 0.600
(1000, 10, 10), (2, 2, 2, 2), cuda, torch.float32, 0.0000226, 0.0000650, 0.348
(1000, 10, 10), (2, 2, 2, 2), cuda, torch.float64, 0.0000328, 0.0000975, 0.336
(1000, 10, 10), (2, 2, 2, 2), cuda, torch.int32, 0.0000369, 0.0001086, 0.340
* (10, 100, 100), (10, 10, 10, 10), cpu, torch.float32, 0.0000410, 0.0002057, 0.199
* (10, 100, 100), (10, 10, 10, 10), cpu, torch.float64, 0.0000471, 0.0002930, 0.161
* (10, 100, 100), (10, 10, 10, 10), cpu, torch.int32, 0.0000411, 0.0002048, 0.201
(10, 100, 100), (10, 10, 10, 10), cuda, torch.float32, 0.0000225, 0.0000652, 0.345
(10, 100, 100), (10, 10, 10, 10), cuda, torch.float64, 0.0000363, 0.0000982, 0.370
(10, 100, 100), (10, 10, 10, 10), cuda, torch.int32, 0.0000331, 0.0000982, 0.337
* (10, 100, 100), (100, 100, 100, 100), cpu, torch.float32, 0.0004513, 0.0010880, 0.415
* (10, 100, 100), (100, 100, 100, 100), cpu, torch.float64, 0.0005202, 0.0011243, 0.463
* (10, 100, 100), (100, 100, 100, 100), cpu, torch.int32, 0.0004035, 0.0009218, 0.438
(10, 100, 100), (100, 100, 100, 100), cuda, torch.float32, 0.0000416, 0.0000734, 0.567
(10, 100, 100), (100, 100, 100, 100), cuda, torch.float64, 0.0000438, 0.0000824, 0.532
(10, 100, 100), (100, 100, 100, 100), cuda, torch.int32, 0.0000416, 0.0000733, 0.568
* (100, 100, 100), (10, 10, 10, 10), cpu, torch.float32, 0.0004485, 0.0025622, 0.175
* (100, 100, 100), (10, 10, 10, 10), cpu, torch.float64, 0.0009544, 0.0016554, 0.577
* (100, 100, 100), (10, 10, 10, 10), cpu, torch.int32, 0.0004403, 0.0016891, 0.261
(100, 100, 100), (10, 10, 10, 10), cuda, torch.float32, 0.0000717, 0.0001155, 0.621
(100, 100, 100), (10, 10, 10, 10), cuda, torch.float64, 0.0000849, 0.0001416, 0.600
(100, 100, 100), (10, 10, 10, 10), cuda, torch.int32, 0.0000717, 0.0001158, 0.619
* (10, 1000, 1000), (10, 10, 10, 10), cpu, torch.float32, 0.0121124, 0.0221824, 0.546
* (10, 1000, 1000), (10, 10, 10, 10), cpu, torch.float64, 0.0228420, 0.0327722, 0.697
* (10, 1000, 1000), (10, 10, 10, 10), cpu, torch.int32, 0.0117615, 0.0204379, 0.575
(10, 1000, 1000), (10, 10, 10, 10), cuda, torch.float32, 0.0003801, 0.0003391, 1.121
(10, 1000, 1000), (10, 10, 10, 10), cuda, torch.float64, 0.0005633, 0.0005638, 0.999
(10, 1000, 1000), (10, 10, 10, 10), cuda, torch.int32, 0.0002959, 0.0002855, 1.036
* (10, 1000, 1000), (100, 100, 100, 100), cpu, torch.float32, 0.0173225, 0.0292359, 0.593
* (10, 1000, 1000), (100, 100, 100, 100), cpu, torch.float64, 0.0324743, 0.0443235, 0.733
* (10, 1000, 1000), (100, 100, 100, 100), cpu, torch.int32, 0.0170375, 0.0292668, 0.582
(10, 1000, 1000), (100, 100, 100, 100), cuda, torch.float32, 0.0005092, 0.0004506, 1.130
(10, 1000, 1000), (100, 100, 100, 100), cuda, torch.float64, 0.0006717, 0.0006754, 0.995
(10, 1000, 1000), (100, 100, 100, 100), cuda, torch.int32, 0.0003871, 0.0003588, 1.079
```

</details>

For CUDA, the compiled performance seems fairly similar with and without the new decomposition. But CPU is about 2x slower in most of the cases I marked, and down to about 5x slower in the worst case